pkg/api/types.go

```diff
@@ -1015,6 +1015,21 @@ func (p *Properties) HasZonesForAllAgentPools() bool {
 	return false
 }
 
+// IsUbuntuDistroForAllAgentPools returns true if all of the agent pools plus masters are running the base Ubuntu image
+func (p *Properties) IsUbuntuDistroForAllAgentPools() bool {
+	if len(p.AgentPoolProfiles) > 0 {
+		for _, ap := range p.AgentPoolProfiles {
+			if ap.Distro != Ubuntu {
+				return false
+			}
+		}
+	}
+	if p.MasterProfile.Distro == Ubuntu {
```

Comment:
this could be `return p.MasterProfile.Distro == Ubuntu`

Reply:
`return p.MasterProfile.Distro == Ubuntu || p.MasterProfile.Distro == Ubuntu1804` actually

Reply:
done